resource-managers/yarn/src/test/java/org/eclipse/jetty/server/SessionManager.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.jetty.server;
+
+import javax.servlet.SessionCookieConfig;
+import javax.servlet.SessionTrackingMode;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.EventListener;
+import java.util.Set;
+
+import org.eclipse.jetty.http.HttpCookie;
+import org.eclipse.jetty.server.session.SessionHandler;
+import org.eclipse.jetty.util.component.LifeCycle;
+
+/**
+ * Adapted from https://github.com/eclipse/jetty.project/blob/jetty-9.3.25.v20180904/
```

Comment:
OK yeah, seems like the least-bad option to get this working.

Reply:
I think if the original file has a license, we can't replace or change it to Apache v2 at the topic.

FYI, jetty eclipse license here https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/LICENSE

Reply:
It will fail when running RAT checks:
```
========================================================================
Running Apache RAT checks
========================================================================
Attempting to fetch rat
Could not find Apache license headers in the following files:
 !????? /home/jenkins/workspace/SparkPullRequestBuilder/resource-managers/yarn/src/test/java/org/eclipse/jetty/server/SessionManager.java
 !????? /home/jenkins/workspace/SparkPullRequestBuilder/resource-managers/yarn/src/test/java/org/eclipse/jetty/server/session/SessionHandler.java
[error] running /home/jenkins/workspace/SparkPullRequestBuilder/dev/check-license ; received return code 1
```

Reply:
If these are a straight copy, then it needs the original source's copyright to be retained as-is, yes, and we can't add an ASF header. You can exclude this file from RAT in `dev/.rat-excludes` (You are welcome to update it to 0.13 too while you're at it.)

Handling of EPL v1.0 is a little more nuanced:
https://apache.org/legal/resolved.html#weak-copyleft-licenses

The notes don't quite cover this case, of test-only code. The logic is that we want to avoid users creating a derivative work of this code. I don't see any case where a downstream user would modify this or include it in a software product; I can imagine someone bundling the Spark test code as-is, but not depending on it.

I'll ask legal if there's more to it.

Reply:
Actually, it's dual licensed under the Apache License 2.0 and EPL v1.0. We need to keep the original header and exclude the RAT check in both files, but there's no other issue here.